src/vcpkg/triplet.cpp

```diff
@@ -83,13 +84,38 @@ namespace vcpkg
         return Triplet::from_canonical_name(std::move(canonical_name));
     }
 
-    Triplet default_triplet(const VcpkgCmdArguments& args)
+    Triplet default_triplet(const VcpkgCmdArguments& args, bool is_manifest_mode)
     {
+        (void)is_manifest_mode;
         if (auto triplet = args.triplet.get())
         {
+            // The triplet is set by --triplet or VCPKG_DEFAULT_TRIPLET.
             return Triplet::from_canonical_name(*triplet);
         }
 #if defined(_WIN32)
+        // Skip warning message on JSON output
+        if (!Strings::case_insensitive_ascii_equals(args.command, "z-print-config"))
+        {
+            // The triplet is not specified or is part of a string qualifier i.e. zlib:x64-windows
+            auto host_triplet = system_triplet().canonical_name();
+            if (is_manifest_mode)
+            {
+                msg::println_warning(msgDefaultTriplet, msg::triplet = host_triplet);
+            }
+            else
+            {
+                for (auto&& arg : args.command_arguments)
```

Comment:
I'm really not a fan of assuming that the command arguments are ports regardless of the executed command like this. For instance, I believe this will do the wrong thing for things like `add artifact`. (That is `vcpkg add artifact gcc` should not emit a warning but I believe this will make it do so)

I think commands that are calling `check_and_get_full_package_spec` need different handling here than commands that do not.